start_jupyter_cm/gnome.py

```diff
@@ -45,13 +45,23 @@ def add_jupyter_here():
              'notebook': os.path.join(logo_path, 'jupyter.png'),
              'lab': os.path.join(logo_path, 'jupyter.png')}
 
+    path_python,filename = os.path.split(shutil.which("python"))
+
     for terminal in ["qtconsole", "notebook", "lab"]:
         script_path = os.path.join(SPATH, "Jupyter %s here%s" % (
                 terminal, CONDA_ENV_LABEL))
         if (not os.path.exists(script_path) and
             shutil.which("jupyter-%s" % terminal)):
+            SYSPATH,filename = os.path.split(shutil.which("jupyter-%s" % terminal))
+            if (SYSPATH == PATH):
+                path_exec = PATH
+            else:
+                print("        NOTE: jupyter-%s lies outside env directory, using this path instead:" % terminal)
```

Comment:
I am not sure if the message is clear enough for the users. I imagine that this situation can arise only when jupyter is not installed in the current environment but is installed in system's python. Wouldn't it be useful to make this more explicit so that the user easily understands that to fix the issue it suffice with installing jupyter in the current environment?

Reply:
This situation can occur when jupyter is installed in the base conda environment, but is also then used in another environment too. In this situation conda doesn't duplicate the installation of jupyter (unless a different version is installed), instead it uses the base conda install.

The message is just for information, and isn't strictly necessary - the installer fixes the problem by simply finding the path to the jupyter executable itself, so the user doesn't have to do anything else. If you think it might be confusing for the user then we could simply remove the message text?

Reply:
Can it also happen in the situation that I mention? If yes, then I think that it will be worth keeping the message because it is unlikely that the user will want to open his system's python jupyter when calling `start_jupyter_cm` from anaconda. If no, then I agree that it'll be better to remove it not to worry users unnecessarily.

Reply:
In theory it could, as it would simply require a system install of jupyter to be available on the executable search path, without any anaconda installed version, in a directory that is higher on the search path list, superceding it.

I could add some text advising the user on how to interpret this message: "Sometimes conda environments will use executables in the base conda install. However, if you do not think this is the correct version of jupyter to use then uninstall the tool, and check your environment settings before reinstalling."

Reply:
Sounds good!

Reply:
Cool - I'll amend the code.

Reply:
I added the code, then found I couldn't replicate the setup in which this code would be needed.

After some experimentation I've realised there's a bug in conda deactivate (I've just submitted a bug report here: https://github.com/conda/conda/issues/9280), which had mislead me as to how conda deals with package management.

I will remove the path substitution code, as it is an error. Would you still like a check for this edge-case error? If not I'll remove that check too, and will just leave the fix for the hardcoded python path in this pull request.

Reply:
So far, nobody has been hit by the corner case that I mention. So, if the issue that you were trying to fix is actually a bug elsewhere, I think that it is better too keep it simple and remove the code as you suggest.

Reply:
I agree. I've removed the code & updated the pull request.